### PR TITLE
Support deploying/destroying ephemeral environments

### PR DESCRIPTION
**Pull request by alexmt**
Closes Issue #41 

Two APIs  were added, changed.

POST /api/v1/applications/ephemeral
```
{
  "source": {
     "repoURL": "...." // repository URL containing the ksonnet application.
     "path": "..." //directory path within repository 
    "environment": "..." // environment name which contains 'ephemeral' application template
    "targetRevision": "..." // repo revision
   },
   "destination": {
       "server": "..." // target cluster URL where app should be deployed
       "namespace": ".." // target namespace
   },
   "dryRun" : false,
   "inputFiles" : { // list of files which should be created in environment directory before generating manifests
        "myParams.libsonnet":  "{ \"test\": false }"
    }
}
```

Ephemeral application might be deleted using API call: 

DELETE /api/v1/applications/<app_name>?namespace=<namespace>&server=<server>

### CODE REVIEW

File: util/kube/kube.go [hunk 1]
```diff
@@ -84,6 +84,40 @@ func GetLiveResource(dclient dynamic.Interface, obj *unstructured.Unstructured,
...
+	resources, err := disco.ServerResources()
+	if err != nil {
+		return err
+	}
+	for _, apiResourcesList := range resources {
+		for _, apiResource := range apiResourcesList.APIResources {
+			deleteSupported := false
+			for _, verb := range apiResource.Verbs {
+				if verb == "deletecollection" {
+					deleteSupported = true
```
Comment (merenbach) [inline util/kube/kube.go]: @alexmt maybe worth adding a `break` here terminate iteration?

**Review by merenbach — APPROVED** [review]
Just one quick question about a possible `break` statement.  Looks good!

File: util/kube/kube.go (hunk 1 above)
Comment (alexmt) [reply util/kube/kube.go]: Done

Comment (merenbach) [reply util/kube/kube.go]: Awesome!  Thank you!

**Pull request merged**